src/satellite/process.ts

```diff
@@ -301,38 +320,67 @@ export class SatelliteProcess implements Satellite {
   }
 
   // Perform a snapshot and notify which data actually changed.
-  async _performSnapshot(): Promise<void> {
-    const oplog = this.opts.oplogTable.toString()
-    const timestamp = new Date().toISOString()
+  async _performSnapshot(): Promise<Date> {
+    const timestamp = new Date()
 
-    const updateTimestamps = `
-      UPDATE ${oplog} set timestamp = ?
-        WHERE rowid in (
-          SELECT rowid FROM ${oplog}
-              WHERE timestamp is NULL
-              AND rowid > ?
-          ORDER BY rowid ASC
-          )
-    `
+    await this._updateOplogTimestamp(timestamp)
+    const oplogEntries = await this._getUpdatedEntries(timestamp)
 
-    const updateArgs = [timestamp, `${this._lastAckdRowId}`]
-    await this.adapter.run({ sql: updateTimestamps, args: updateArgs })
+    const promises: Promise<void | SatelliteError | RunResult>[] = []
 
-    const selectChanges = `
-      SELECT * FROM ${oplog} 
-      WHERE timestamp = ? ORDER BY rowid ASC
-    `
+    if (oplogEntries.length !== 0) {
+      // FIXME: We should probably notify about changes later
+      promises.push(this._notifyChanges(oplogEntries))
 
-    const rows = await this.adapter.query({
-      sql: selectChanges,
-      args: [timestamp],
-    })
-    const results = rows as unknown as OplogEntry[]
+      const shadowEntries = new Map<string, ShadowEntry>()
```

Comment:
So as far as I can tell, `shadowEntries` is cached only in the scope of this function call, i.e. when the same row is touched multiple times in the same oplog "span", right?

Reply:
Yes, that's the idea.